### ISSUE TITLE
Travel Pay SMOC: Accessibility Testing

### ISSUE THREAD

**Issue by kjduensing** [department-of-veterans-affairs/va.gov-team#100924]
Platform guidance: https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review

### Product Information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [ ] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test? 
Calvin Cieslak (@seesleestak) tested with the Chrome browser for both desktop and mobile (Android).
Calvin Cieslak (@seesleestak) tested with the Safari browser for both desktop (Macbook Air) and mobile (iPhone 13 Pro).

### Use of color and color contrast checks
Tools used: Wave and Axe extensions for Chrome, Colour Contrast Analyzer for Mac.

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1 contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

All elements within the the SMOC flow pass WCAG color contrast criteria

### Who completed the axe scans?
Calvin Cieslak (@seesleestak)

Axe Checks:

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### How did axe checks go?

All elements within the the SMOC flow pass axe checks

### Who completed the content zoom and reflow test?
Calvin Cieslak (@seesleestak) tested with the Chrome browser for both desktop and mobile (Android).
Calvin Cieslak (@seesleestak) tested with the Safari browser for both desktop (Macbook Air) and mobile (iPhone 13 Pro).

### Content zoom and reflow checks (desktop AND mobile)
**Desktop:**

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

Successfully tested on desktop (Chrome and Safari) and all page elements are readable and usable at 200, 300, and 400% zoom.

The following screenshots are in Chrome. Safari was also tested for all cases but had less robust screenshot support.

|  | Entry point | Mileage | Vehicle | Address | Review | Confirmation | "No" selection |
|--------|---------|----------|----------|----------|---------|--------------|-----|
| 200% | ![Image](https://github.com/user-attachments/assets/b51dd248-99b8-440a-b8f0-892756ba445b) | ![Image](https://github.com/user-attachments/assets/6732f595-ecf3-432f-a659-fa38af4c52ca) | ![Image](https://github.com/user-attachments/assets/33b52ece-45c1-4c89-bfeb-80d888101ec4) | ![Image](https://github.com/user-attachments/assets/71c54945-36e8-4162-8e65-5463f483e36c) | ![Image](https://github.com/user-attachments/assets/40e3f01f-7228-442f-898b-f7eb1c4ebc9e) | ![Image](https://github.com/user-attachments/assets/a8e47b22-4822-4d5d-ad29-95b091b18b9e) | ![Image](https://github.com/user-attachments/assets/b17029d2-d76d-4686-97c4-90ae31e5c821) |
| 300% | ![Image](https://github.com/user-attachments/assets/2d030df0-dfa3-4f30-a77b-935ca0ca5431) | ![Image](https://github.com/user-attachments/assets/5f8e6876-d88c-4f58-9851-03e7188ff2af) | ![Image](https://github.com/user-attachments/assets/f46b202e-5fcc-4d45-b92b-3e3a79c39289) | ![Image](https://github.com/user-attachments/assets/3336ee7c-7a9c-4b5c-99f9-553025cc5434) | ![Image](https://github.com/user-attachments/assets/9eef31fc-ffc1-4976-a904-fbfcfbc8cb9d) | ![Image](https://github.com/user-attachments/assets/da1583b9-44ed-4fd6-8eb7-b2bf2dcbd650) | ![Image](https://github.com/user-attachments/assets/1f745fa7-9ea4-4399-aff7-f2b9a6296701) |
| 400% | ![Image](https://github.com/user-attachments/assets/1b6099d4-73d0-43bf-8bfb-28be9d7ba76d) | ![Image](https://github.com/user-attachments/assets/241337e4-2139-488d-9b78-4c50104bd4c0) | ![Image](https://github.com/user-attachments/assets/2eb3fdd0-d32e-416c-bc1e-30aa5d1b7ef4) | ![Image](https://github.com/user-attachments/assets/39f8f42e-3192-4d08-a196-4d1ca4869100) | ![Image](https://github.com/user-attachments/assets/cbfca7e3-dff9-4552-a3be-097f4ed15f6d) | ![Image](https://github.com/user-attachments/assets/9de53641-8415-4e68-b61d-7d459ddd1dd3) | ![Image](https://github.com/user-attachments/assets/ad466e1f-25a8-4ab9-a282-b5d3bd2c8166) |

**Mobile:**

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom Chrome on Android and Safari on iPhone both have a limit of 300%.

For both mobile browsers, many page elements are not readable and usable at 200% and 300% zoom. This includes:

- Issues with text not wrapping / getting cut off
- Issues with resizing
- Issues with content taking up the entire screen (Safari on iPhone at 300%)

More details, including screenshots, in the comments below.

### Who completed the keyboard navigation test?
Calvin Cieslak (@seesleestak) tested with the Chrome browser.
Calvin Cieslak (@seesleestak) tested with the Safari browser.

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

All elements within the the SMOC flow pass the above criteria

### [Screen Reader testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/screen-reader-testing)

The accessibility specialists are encouraging front-end engineers to do basic screen reader testing as part of their software development cycle.

- https://a11yproject.com/posts/macos-browser-keyboard-navigation/
- https://help.apple.com/voiceover/info/guide/10.12

### Do you have any other results to share?

**Comment by seesleestak** [comment]
@kjduensing For mobile testing, is using the desktop browser's mobile simulation acceptable or do I need to use something like `ngrok` to get a local build of SMOC on my phone for testing?

**Comment by seesleestak** [comment]
I've gone through all of the accessibility criteria in this ticket and found no issues. I've added screenshots to the ticket where pertinent.
